Suggested template for room rules and reg

Chiamakabrowneyes opened the pull request:
I made this as a chore list, but since that exists, we could revise it.
Since i'm having issues with creating the profile, can some one help me add this code to the route. thanks

Commit:
help review the blank contact me section

Commit:
help review the blank contact me section

Commit:
help review the blank contact me section

Commit:
help review the blank contact me section

Commit:
help review the blank contact me section

Commit:
help review the blank contact me section

Commit:
template for rules

Commit:
template for rules

Commit:
template for rules

Commit:
template for rules

Commit:
template for rules

Commit:
template for rules

Commit:
template for rules

Commit:
template for rules

Commit:
template for rules

Commit:
template for rules

Commit:
template for rules

Commit:
template for rules

Commit:
template for rules

Commit:
template for rules

Commit:
template for rules

Commit:
adding template back

Commit:
adding template back

Commit:
adding template back

Commit:
adding template back

Commit:
adding template back

Commit:
adding template back

Commit:
adding template back

Commit:
adding template back

Commit:
adding template back

Commit:
adding template back

Commit:
adding template back

Commit:
adding template back

Commit:
adding template back

Commit:
adding template back

Commit:
adding template back

bathientran commented:
@Chiamakabrowneyes the PR only shows removed files, what exactly should be viewed here? :)

Chiamakabrowneyes commented:
> @Chiamakabrowneyes the PR only shows removed files, what exactly should be viewed here? :)

wait really? 
I had added and removed and added again. Let me check

vickipetrova committed:
Many small frontend improvement

vickipetrova committed:
Many small frontend improvement

vickipetrova committed:
fixed

vickipetrova committed:
Add the new license key and add it to the env variables

bathientran committed:
fix(): adds NEXT_PUBLIC prefix for public env variable

bathientran committed:
fix(): fixes logic to check for chorelists and default image rendering

bathientran committed:
feat(): moves settings into bottom bar and changes fixes icon usage

bathientran committed:
feat(): updates announcements styling

bathientran committed:
feat(): updates top bar to be sticky and main body to offset top bar

bathientran committed:
fix(): adjusts settings page to offset top sticky bar

vickipetrova committed:
Added dark mode

vickipetrova committed:
Merge remote-tracking branch 'origin/frontend-omprovements' into frontend-omprovements

vickipetrova committed:
Added dark mode and made top nav bar float like bottom nav bar.

bathientran committed:
fix(announcements): fix case when member does not exist for announcement

bathientran committed:
fix(announcements): adds conditional for announcement member name as well

vickipetrova committed:
Merge pull request #43 from minerva-startup-spring2024/frontend-omprovements

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

richiewaweru committed:
tessting deployment on new branch

richiewaweru committed:
tessting deployment on new branch

vickipetrova committed:
Updated readme instructions

bathientran committed:
docs(readme): added a table of contents and adjusted some of the doc a bit

vickipetrova committed:
Merge pull request #46 from minerva-startup-spring2024/readme

Updated readme instructions

richiewaweru committed:
added full range building management capability

richiewaweru committed:
fixed room connection errors

minerva-startup-spring2024 committed:
Merge pull request #44 from minerva-startup-spring2024/add-calendar-license

Add the new license key and add it to the env variables

Commit:
adding cards

Commit:
adding cards

Commit:
adding cards

Commit:
adding cards

Commit:
adding cards

Commit:
adding cards

Commit:
adding cards

Commit:
adding cards

Commit:
adding cards

Commit:
adding cards

bathientran committed:
feat(calendar): fixes calendar view

bathientran committed:
feat(): removes app.css

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

Commit:
Completion for the rules section

bathientran committed:
infra(migrations): fixes migrations

bathientran committed:
chore(routes): updates routes imports to work for both room and building membership

bathientran committed:
chore(routes): fixes buildings routes

bathientran committed:
chore(TopBar): updates top bar to show both rooms and buildings

bathientran committed:
chore(): refactors overviews and creation components to work for both buildings and rooms, adjusts old getprofile calling to match new generalized function

bathientran committed:
chore(): missing from last commit

bathientran committed:
chore(): updates styling and padding

bathientran committed:
chore(): add error logging for room retrieval

bathientran committed:
chore(): makes memberlist retrieval generalized

bathientran committed:
chore(building): adds all building components and rendering

waiyego4 committed:
logout functionality and small UI changes

minerva-startup-spring2024 committed:
Merge pull request #50 from minerva-startup-spring2024/feat-building

Feat building

bathientran committed:
chore(): fixes conflicts and makes topbar client side

minerva-startup-spring2024 committed:
Merge pull request #51 from minerva-startup-spring2024/log-out

logout functionality and small UI changes

vickipetrova committed:
fix() edit action - it failed because we didn't have the correct event id. The bug happened when you add a new event -> try to edit it without reloading the page. It's because we only fetch the events when the page loads. So I'll just add a line to fetch all events right after the creation of a new event.

vickipetrova committed:
Added some padding to the sides of the calendar so it doesn't stretch to the edges of the screen

vickipetrova committed:
Merge branch 'staging' into GENERAL-94-fix-calendar-fronted

vickipetrova committed:
hotfix(): fixed an error introduced by this commit https://github.com/minerva-startup-spring2024/roommate-organizer/commit/6a7e3a147ca171e4f2527f2343d23364e9414c98. The app failed to run with this error: Cannot select both '$scalars: true' and a specific scalar field 'members'. ⨯ PrismaClientUnknownRequestError: Invalid `prisma.profile.findUnique()` invocation: Cannot select both '$scalars: true' and a specific scalar field 'members'. at async getProfile (./src/app/api/_utils.js:12:17). I think it's because the database schema hasn't been updated yet, we don't have a relation table between the buildings and members and we don't have a relationship defined between the buildings and rooms yet. For now I just commented out the lines causing the error

vickipetrova committed:
Merge pull request #52 from minerva-startup-spring2024/fix-bug

hotfix(): previous commit broke the app

vickipetrova committed:
Merge branch 'staging' into GENERAL-94-fix-calendar-fronted

vickipetrova committed:
Merge pull request #48 from minerva-startup-spring2024/GENERAL-94-fix-calendar-fronted

feat(calendar): fixes calendar view

vickipetrova committed:
Trying to make the recurring events work

vickipetrova committed:
Events can be recurring - the database is updated in accordance with that.

vickipetrova committed:
Added to the events API documentation

vickipetrova committed:
Added to the events API documentation

vickipetrova committed:
Documentation fix

vickipetrova committed:
fix merge

vickipetrova committed:
make a single migration file

vickipetrova committed:
make a single migration file as it is by default

vickipetrova committed:
Initial integration with Canny

vickipetrova committed:
Fixed the Canny token generation

vickipetrova committed:
hotfix(): attempt to fix db by manually changing the migration file, removed anything related to buildings since it shouldn't be here

bathientran committed:
chore(announcements): changes how announcements are displayed and adds view all option

bathientran committed:
chore(announcements): changes how announcements are displayed and adds view all option

bathientran committed:
chore(announcements): refactors announcement implementation

bathientran committed:
chore(announcements): refactors announcement implementation

bathientran committed:
chore(announcements): refactors announcement implementation

bathientran committed:
chore(feedback): removes empty page container

bathientran committed:
Merge pull request #54 from minerva-startup-spring2024/collect-user-feedback

Collect user feedback

bathientran committed:
chore(dark-mode): remove dark mode for now

bathientran committed:
chore(dark-mode): fix missing commenting

minerva-startup-spring2024 committed:
Merge pull request #55 from minerva-startup-spring2024/HOTFIX-remove-dark-mode

Hotfix remove dark mode

richiewaweru committed:
iplmented manager to room sending of messages

richiewaweru committed:
iplmented manager to room sending of messages

richiewaweru committed:
implmented roommate to roommate communication and roommate to manager communication

richiewaweru committed:
implmented roommate to roommate communication and roommate to manager communication

richiewaweru committed:
APIs to allow deletion and updating and created a simple frontend for manager's messages viewing

richiewaweru committed:
APIs to allow deletion and updating and created a simple frontend for manager's messages viewing

richiewaweru committed:
fix utils import error

richiewaweru committed:
fix utils import error

minerva-startup-spring2024 committed:
Merge pull request #56 from minerva-startup-spring2024/enhance-calendar-functionalities

Enhance calendar functionalities

richiewaweru committed:
corect the production error

richiewaweru committed:
corect the production error

Commit:
conflict..

Commit:
conflict..

Commit:
conflict..

richiewaweru committed:
remove frontend from commit

richiewaweru committed:
remove frontend from commit

richiewaweru committed:
removed frontend implmentation

richiewaweru committed:
removed frontend implmentation

richiewaweru committed:
removed frontend implenentation

richiewaweru committed:
removed frontend implenentation

bathientran committed:
hotfix(): fixes inconsistent migrationand schema setup

minerva-startup-spring2024 committed:
Merge pull request #59 from minerva-startup-spring2024/HOTFIX-resetting-db

hotfix(): fixes inconsistent migrationand schema setup

bathientran committed:
fix(announcements-be): removes redundant component and fixes schema

bathientran committed:
infra(announcements-be): apply migration and get up to date with staging

bathientran committed:
fix(announcements-be): simplifies announcements creation route

bathientran committed:
fix(): remove empty line from topbar

bathientran committed:
fix(): fixes typeo and changes deconstruction in announcement creation to use data key

bathientran committed:
fix(): removes building routes temporarily

minerva-startup-spring2024 committed:
Merge pull request #57 from minerva-startup-spring2024/richiefeatannouncemnts

Added building announcement and modiified user api to allow back and forth communication between roommates and managers

bathientran committed:
fix(announcements-fe): fixes PR structure

bathientran committed:
chore(announcements-fe): replace frontend access to announcements in rooms

bathientran committed:
chore(announcements-fe): removes legacy component

bathientran committed:
chore(announcements-be): fixes backend logic for creating an announcmenet with and without a receiver

bathientran committed:
chore(): replaces message creation frontend

bathientran committed:
chore(): moves add message to top

minerva-startup-spring2024 committed:
Merge pull request #49 from minerva-startup-spring2024/adding-announcement

Adding announcement

minerva-startup-spring2024 committed:
Merge pull request #60 from minerva-startup-spring2024/staging

DEPLOY 0.0.1

Chiamakabrowneyes committed:
Merge branch 'main' of github.com:minerva-startup-spring2024/roommate-organizer into adding_ruls

Chiamakabrowneyes committed:
adding backend for rules

Chiamakabrowneyes committed:
adding backend for rules

Chiamakabrowneyes committed:
fixing error

Chiamakabrowneyes committed:
fixing error

Chiamakabrowneyes committed:
added forgot password functionality

Chiamakabrowneyes committed:
added forgot password functionality

Chiamakabrowneyes committed:
making changes to view

Chiamakabrowneyes committed:
making changes to view

Chiamakabrowneyes committed:
Merge branch 'adding-ruls' of github.com:minerva-startup-spring2024/roommate-organizer into forgot-password

Chiamakabrowneyes committed:
Merge branch 'adding-ruls' of github.com:minerva-startup-spring2024/roommate-organizer into adding-ruls

Chiamakabrowneyes committed:
merging most recent changes to mobile view
Merging current changes for rules in mobile viewbranch 'forgot-password' of github.com:minerva-startup-spring2024/roommate-organizer into adding-ruls

Chiamakabrowneyes committed:
adapting the mobile view for rules function

Chiamakabrowneyes committed:
adapting the mobile view for rules function

Chiamakabrowneyes committed:
adding mobile func

Chiamakabrowneyes committed:
adding mobile func

Chiamakabrowneyes committed:
attempting to fix migration error by deleting migration folder locally

Chiamakabrowneyes committed:
attempting to fix migration error by deleting migration folder locally

Chiamakabrowneyes committed:
cleaning up the unneccessary code

Chiamakabrowneyes committed:
cleaning up the unneccessary code

bathientran committed:
chore(): adds migrations back

Chiamakabrowneyes committed:
adding updates to branch
Merge branch 'adding-ruls' of github.com:minerva-startup-spring2024/roommate-organizer into adding-ruls

Chiamakabrowneyes committed:
cleaning up the unneccessary code

Chiamakabrowneyes committed:
cleaning up the unneccessary code

Chiamakabrowneyes committed:
adding back the migration folder

bathientran committed:
chore(): adds back the .env.local file

Chiamakabrowneyes committed:
Merge pull request #45 from minerva-startup-spring2024/adding-ruls

Adding rules

bathientran committed:
fix(): fixes image loading issues

minerva-startup-spring2024 committed:
Merge pull request #65 from minerva-startup-spring2024/HOTFIX-images

fix(): fixes image loading issues

minerva-startup-spring2024 committed:
Merge pull request #66 from minerva-startup-spring2024/staging

deploy V0.0.2

bathientran committed:
chore(): removes password strength test

minerva-startup-spring2024 committed:
Merge pull request #67 from minerva-startup-spring2024/HOTFIX-password-strength

chore(): removes password strength test

minerva-startup-spring2024 committed:
Merge pull request #68 from minerva-startup-spring2024/staging

deploy: V0.0.3

vickipetrova committed:
Fixed the profile image not loading on the announcements

vickipetrova committed:
Streamlined the style by making the room preview page have the same background color and other small styling changes

vickipetrova committed:
Added a label for the roommates button, hopefully it's more clear this way

vickipetrova committed:
Changed the string for the chores/shopping items from My to created by me so it's more descriptive

vickipetrova committed:
Merge pull request #69 from minerva-startup-spring2024/user-feedback-improvements

User feedback improvements

minerva-startup-spring2024 committed:
Merge pull request #70 from minerva-startup-spring2024/staging

deploy: 0.0.4

bathientran committed:
fix(api): adding error logging

bathientran committed:
Merge branch 'staging'

bathientran committed:
fix(api): adds prisma back

bathientran committed:
Merge branch 'staging'
